gammapy/modeling/models/core.py

```diff
@@ -681,6 +681,30 @@ def copy(self, copy_data=False):
             models=models, covariance_data=self.covariance.data.copy()
         )
 
+    def slice_by_energy(self, energy_min, energy_max, sum_over_energy_groups=False):
```

Comment:
A naive question: this function slices only the `TemplateNPredModel'. Should we have a more explicit function name?

Reply:
I would says no to be consistent with dataset.slice_by_energy